docs/data/material/customization/container-queries/container-queries.md

```diff
@@ -0,0 +1,86 @@
+# Container Queries
+
+<p class="description">Material UI provides a utility function for creating CSS container queries based on theme breakpoints.</p>
+
+## Usage
+
+To create [CSS container queries](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries), use `theme.containerQueries` with any method available in the [`theme.breakpoints`](/material-ui/customization/breakpoints/#api).
+The value can be unitless (in which case it'll be rendered in pixels), a string, or a breakpoint key. For example:
+
+```js
+theme.containerQueries.up('sm'); // => '@container (min-width: 600px)'
+```
+
+:::info
+One of the ancestors must have the CSS container type specified.
+:::
+
+{{"demo": "BasicContainerQueries.js"}}
+
+### Named containment contexts
+
+To refer to a [containment context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries#naming_containment_contexts), call the `containerQueries` method with the name of the container for access to all breakpoint methods:
+
+```js
+theme.containerQueries('sidebar').up('500px'); // => '@container sidebar (min-width: 500px)'
+```
+
+## Shorthand syntax
+
+When adding styles using the `sx` prop, use the `@<size>` or `@<size>/<name>` notation to apply container queries without referring to the theme.
+
+- `<size>`: a width or a breakpoint key.
+- `<name>` (optional): a named containment context.
+
+For example:
+
+```js
+<Card
+  sx={{
+    display: 'flex',
+    flexDirection: {
+      '@': 'column', // @container (min-width: 0px)
+      '@250': 'row', // @container (min-width: 250px)
+    },
+  }}
+/>
+```
```

Comment:
Do we need this code block, given this code is already in the demo below? Like, can't we have the uncollapsed piece of code from the demo show just this part?

Reply:
I can remove it.